### DISCUSSION TITLE
como executar esses comandos ao iniciar no i3wm sem pedir a senha? Can anyone help me how to run commands when starting in i3wm without asking for the password?

### DISCUSSION THREAD

**Discussion by ronibastos** [i3/i3#6014] [Q&A] [answered]
vim .config/i3/config

exec --no-startup-id alacritty -e sudo apt autoremove
exec --no-startup-id alacritty -e sudo apt autoclean
exec --no-startup-id alacritty -e sudo apt install -f
exec --no-startup-id alacritty -e sudo apt update && -e sudo apt upgrade
exec --no-startup-id alacritty -e sudo apt update
exec --no-startup-id alacritty -e sudo apt upgrade
exec --no-startup-id alacritty -e sudo apt full-upgrade

como executar esses comandos ao iniciar no i3wm sem pedir a senha?
Can anyone help me how to run commands when starting in i3wm without asking for the password?

**Comment by okraits** [comment]
You need to configure sudo to not ask for a password. Hint: use `NOPASSWD` in `/etc/sudoers`.

**Reply by ronibastos** [reply]
I'm a beginner and I think I did it wrong! I added it like this /etc/sudoers:

# do not ask for password
NOPASSWD

That didn't work!

**Reply by waterkip** [reply, marked as answer]
You can use this boilerplate. I allow users in the sudoers group to do safe operations without a password. There is a potentially unsafe bit with dist-upgrades (full-upgrades), but since I feel my people have knowledge, I also allow this to be passwordless. Anything else requires a password. Feel free to change it to your liking.

```
# Place this in a file in /etc/sudoers.d/002-apt
#
# Allow apt to run without sudo password for some common operations
Cmd_Alias APT_UPDATE = /usr/bin/aptitude update, /usr/bin/apt-get update, /usr/bin/apt update, /usr/bin/apt-file update
Cmd_Alias APT_UPGRADE = /usr/bin/aptitude safe-upgrade, /usr/bin/apt-get upgrade, /usr/bin/apt upgrade, /usr/bin/aptitude safe-upgrade --no-new-installs

# These are potentially removing things, but are generally safe and needed for upgrades
Cmd_Alias APT_DUPGRADE = /usr/bin/aptitude full-upgrade, /usr/bin/apt-get dist-upgrade, /usr/bin/apt full-upgrade
Cmd_Alias APT_CLEAN = /usr/bin/aptitude clean, /usr/bin/apt-get clean, /usr/bin/apt clean
Cmd_Alias APT_AUTOCLEAN = /usr/bin/aptitude autoclean, /usr/bin/apt-get autoclean, /usr/bin/apt autoclean
Cmd_Alias APT_AUTOREMOVE = /usr/bin/aptitude autoremove, /usr/bin/apt-get autoremove, /usr/bin/apt autoremove

%sudo   ALL=(ALL:ALL) NOPASSWD: APT_UPDATE
%sudo   ALL=(ALL:ALL) NOPASSWD: APT_UPGRADE
%sudo   ALL=(ALL:ALL) NOPASSWD: APT_DUPGRADE
%sudo   ALL=(ALL:ALL) NOPASSWD: APT_CLEAN
%sudo   ALL=(ALL:ALL) NOPASSWD: APT_AUTOCLEAN
%sudo   ALL=(ALL:ALL) NOPASSWD: APT_AUTOREMOVE

```

**Reply by ronibastos** [reply]
Perfect! Everything worked!
This material you sent is incredible!

Thank you very much!!

**Reply by ronibastos** [reply]
> Você pode usar este boilerplate. Eu permito que os usuários do grupo sudoers façam operações seguras sem uma senha. Há um pouco potencialmente inseguro com dist-upgrades (full-upgrades), mas desde que eu sinto que meu pessoal tem conhecimento, eu também permito que isso seja sem senha. Qualquer outra coisa requer uma senha. Sinta-se livre para mudar isso ao seu gosto.
> 
> ```
> # Place this in a file in /etc/sudoers.d/002-apt
> #
> # Allow apt to run without sudo password for some common operations
> Cmd_Alias APT_UPDATE = /usr/bin/aptitude update, /usr/bin/apt-get update, /usr/bin/apt update, /usr/bin/apt-file update
> Cmd_Alias APT_UPGRADE = /usr/bin/aptitude safe-upgrade, /usr/bin/apt-get upgrade, /usr/bin/apt upgrade, /usr/bin/aptitude safe-upgrade --no-new-installs
> 
> # These are potentially removing things, but are generally safe and needed for upgrades
> Cmd_Alias APT_DUPGRADE = /usr/bin/aptitude full-upgrade, /usr/bin/apt-get dist-upgrade, /usr/bin/apt full-upgrade
> Cmd_Alias APT_CLEAN = /usr/bin/aptitude clean, /usr/bin/apt-get clean, /usr/bin/apt clean
> Cmd_Alias APT_AUTOCLEAN = /usr/bin/aptitude autoclean, /usr/bin/apt-get autoclean, /usr/bin/apt autoclean
> Cmd_Alias APT_AUTOREMOVE = /usr/bin/aptitude autoremove, /usr/bin/apt-get autoremove, /usr/bin/apt autoremove
> 
> %sudo   ALL=(ALL:ALL) NOPASSWD: APT_UPDATE
> %sudo   ALL=(ALL:ALL) NOPASSWD: APT_UPGRADE
> %sudo   ALL=(ALL:ALL) NOPASSWD: APT_DUPGRADE
> %sudo   ALL=(ALL:ALL) NOPASSWD: APT_CLEAN
> %sudo   ALL=(ALL:ALL) NOPASSWD: APT_AUTOCLEAN
> %sudo   ALL=(ALL:ALL) NOPASSWD: APT_AUTOREMOVE
> ```

Hello friend, how are you?
A while ago you helped me with this script above, and everything worked out! Now I have the Arch Linux distribution, and I tried to use it but it didn't work.

Could you please help me again now for Arch Linux? I would like to do away with the password in these commands below:

exec --no-startup-id alacritty -e sudo pacman -Scc --noconfirm
exec --no-startup-id alacritty -e sudo pacman -Syyuu --noconfirm

**Reply by waterkip** [reply]
Just use pacman instead of apt and supply the flags with each command? Should not be that hard to adapt.

**Comment by ronibastos** [comment]
Problem resolved successfully!

I would like to thank my colleagues who were willing to help with this matter!

Gratitude!!

**Comment by ronibastos** [comment]
ok thanks for your attention in helping!! thanks


 

    Em terça-feira, 5 de novembro de 2024 às 18:17:05 BRT, Wesley Schwengle ***@***.***> escreveu:  
 
 


Just use pacman instead of apt and supply the flags with each command? Should not be that hard to adapt.
